sys/kern/subr_intr.c

```diff
@@ -1556,10 +1589,10 @@ intr_pic_init_secondary(void)
 	/*
 	 * QQQ: Only root PIC is aware of other CPUs ???
 	 */
-	KASSERT(intr_irq_root_dev != NULL, ("%s: no root attached", __func__));
+	KASSERT(intr_irq_root_dev() != NULL, ("%s: no root attached", __func__));
 
 	//mtx_lock(&isrc_table_lock);
-	PIC_INIT_SECONDARY(intr_irq_root_dev);
+	PIC_INIT_SECONDARY(intr_irq_root_dev());
 	//mtx_unlock(&isrc_table_lock);
```

Comment:
I'm rather concerned about this choice, but the case I'm thinking of may be unusual and not apply here.  I'm unfamiliar with the design of this new Apple hardware and I don't know how this is expected to map to future devices.  I suspect this should instead be looping over all the root PICs and calling the function on each of them in turn?

Reply:
Yeah you have a good point, this happened to work for my use case but it's probably not what we want to do. I'm not sure if just looping over the root PICs is the correct approach either though. This PR allows one device to claim multiple PIC roots and so if we're going to call `pic_init_secondary` multiple times with the same `device_t` it should at least know which root it's for.

Reply:
I added a loop over root PIC devices and modified `pic_init_secondary` to take a `uint32_t root` argument as well. I also replaced the assert for root dev != NULL with just a NULL check, but it might be good to make sure that `pic_init_secondary` is called for at least one device.

@jrtc27 @strejda I pushed the change as a new commit assuming that makes the diff easier to review and plan on squashing the related commits before this is ready to land. Alternatively I'm happy to combine the commits now if you prefer, whatever works for you.